pallets/messages/src/lib.rs

```diff
@@ -146,6 +149,21 @@ pub mod pallet {
 			let msa_id = T::AccountProvider::get_msa_id(&who);
 			ensure!(msa_id.is_some(), Error::<T>::InvalidMessageSourceAccount);
 
+			let message_sender_msa = msa_id.unwrap();
+			match on_behalf_of {
+				Some(producer) => {
+					let current_provider = Provider(message_sender_msa);
+					let current_delegator = Delegator(producer);
+					let provider_info = T::AccountProvider::get_provider_info_of(
+						current_provider,
+						current_delegator,
+					);
+					// TODO: check for specific permissions, such as if message schema
```

Comment:
remove this comment and create an issue.

Reply:
#140